### ISSUE TITLE
GPT2 LoRA _find_and_replace raise tensor shape 0 error with DeepSpeed CPU offloading

### ISSUE THREAD

**Issue by dumpmemory** [huggingface/peft#153]
my accelerate yaml is 

```yaml
compute_environment: LOCAL_MACHINE
deepspeed_config:
  gradient_accumulation_steps: 1
  gradient_clipping: 0.7
  offload_optimizer_device: cpu
  offload_param_device: cpu
  zero3_init_flag: True
  zero3_save_16bit_model: true
  zero_stage: 3
distributed_type: DEEPSPEED
downcast_bf16: 'no'
dynamo_backend: 'NO'
fsdp_config: {}
machine_rank: 0
main_training_function: main
megatron_lm_config: {}
mixed_precision: 'no'
num_machines: 1
num_processes: 8
rdzv_backend: static
same_network: true
use_cpu: false
```

reproduce file 

```python
import os
os.environ['HF_DATASETS_CACHE']="/nfs-nlp-ali/ray_train/huggingface_cache"
os.environ['TRANSFORMERS_CACHE']='/nfs-nlp-ali/ray_train/huggingface_cache/hub'
from transformers import AutoTokenizer, AutoModelForCausalLM
from peft import get_peft_model, LoraConfig, TaskType
from accelerate import Accelerator, DistributedType
from accelerate.logging import get_logger
from accelerate.utils import set_seed
peft_config = LoraConfig(
  task_type=TaskType.CAUSAL_LM,
  inference_mode=False,
  r=16,
  lora_alpha=32,
  lora_dropout=0.1,
)
accelerator = Accelerator(gradient_accumulation_steps=1)
tokenizer = AutoTokenizer.from_pretrained("gpt2-xl")
model = AutoModelForCausalLM.from_pretrained("gpt2-xl")
model = get_peft_model(model, peft_config)
accelerator.print(model.print_trainable_parameters())
```

run command 

```bash
accelerate launch --config_file ds_zero3_cpu.yaml test.py
```
error output
```bash
Traceback (most recent call last):
  File "test.py", line 19, in <module>
    model = get_peft_model(model, peft_config)
  File "/opt/miniconda/lib/python3.8/site-packages/peft/mapping.py", line 145, in get_peft_model
    return MODEL_TYPE_TO_PEFT_MODEL_MAPPING[peft_config.task_type](model, peft_config)
  File "/opt/miniconda/lib/python3.8/site-packages/peft/peft_model.py", line 512, in __init__
    super().__init__(model, peft_config)
  File "/opt/miniconda/lib/python3.8/site-packages/peft/peft_model.py", line 80, in __init__
    self.base_model = LoraModel(peft_config, model)
  File "/opt/miniconda/lib/python3.8/site-packages/peft/tuners/lora.py", line 118, in __init__
    self._find_and_replace()
  File "/opt/miniconda/lib/python3.8/site-packages/peft/tuners/lora.py", line 163, in _find_and_replace
    in_features, out_features = target.weight.shape
```

**Comment by zlucia** [comment]
I'm running into this error as well. The shape attribute only returns 1 value, even though 2 are expected.

**Comment by pacman100** [comment]
@dumpmemory, thanks for the deep dive and the fix for the shape mismatch when using DS Z3 with Conv1D layers. I have included it in PR #157. 

When I use latest DeepSpeed 0.8.2 version, I see no errors when running the code in issue description. Also, it works for fp16 too without any other changes. @dumpmemory  and @zlucia, could you please update to  the latest version of DeepSpeed and let us know if the above PR fixes the issue?
